data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataimport/dao/ScalarDbDao.java

```diff
@@ -248,6 +249,34 @@ public Scanner createScanner(
     }
   }
 
+  /**
+   * Create a ScalarDB scanner instance
+   *
+   * @param namespace ScalarDB namespace
+   * @param table ScalarDB table name
+   * @param projectionColumns List of column projection to use during scan
+   * @param limit Scan limit value
+   * @param transaction Distributed transaction object
+   * @return ScalarDB Scanner object
+   * @throws ScalarDbDaoException if scan fails
+   */
+  public TransactionCrudOperable.Scanner createScanner(
+      String namespace,
+      String table,
+      List<String> projectionColumns,
+      int limit,
+      DistributedTransaction transaction)
+      throws ScalarDbDaoException {
+    Scan scan =
+        createScan(namespace, table, null, null, new ArrayList<>(), projectionColumns, limit);
+    try {
+      return transaction.getScanner(scan);
```

Comment:
Please use `DistributedTransactionManager.getScanner()`. You don’t need to begin or commit a transaction when using it.